Assets/MixedRealityToolkit.Examples/Demos/Input/Scenes/PrimaryPointer/PrimaryPointerHandlerExample.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine;
+
+// Simple example script that subscribes to primary pointer changes and applies a cursor highlight to the current one.
+public class PrimaryPointerHandlerExample : MonoBehaviour
+{
+    public GameObject CursorHighlight;
+
+    private void OnEnable()
+    {
+        MixedRealityToolkit.InputSystem?.FocusProvider?.SubscribeToPrimaryPointerChanged(OnPrimaryPointerChanged, true);
+    }
+
+    private void OnPrimaryPointerChanged(IMixedRealityPointer oldPointer, IMixedRealityPointer newPointer)
+    {
+        if (CursorHighlight != null)
+        {
+            if (newPointer != null)
```

Comment:
A null check is insufficient to detect if the pointer is null. You need to cast it to a monobehavior first, or else it might not be null, but throw a missing reference exception when you try using it

Reply:
A null check should be fine here as we know we're only dealing with pointers that are valid, i.e. registered with the focus provider.

Reply:
Interesting. I guess since you're not caching pointers for more than a frame anyway its not a big issue. I don't love the overhead of adding an update loop that needs to process all the pointers every single update. This is where preventing pointers from being destroyed could be very helpful in keeping complexity per update down.

Reply:
We're caching pointers for more than one frame. We add them to the pointerInfo dictionary when they're registered with the focus provider and remove them when they're unregistered.